setup.* files are in prefix/etc/ros2 instead of prefix

--- ament_package/template/prefix_level/setup.sh.in~	2022-01-18 19:41:21.000000000 +0100
+++ ament_package/template/prefix_level/setup.sh.in	2022-03-30 17:48:51.729340793 +0200
@@ -119,7 +119,8 @@
 if [ "$AMENT_SHELL" = "zsh" ]; then
   ament_zsh_to_array _UNIQUE_PREFIX_PATH
 fi
-for _path in $_UNIQUE_PREFIX_PATH; do
+for _dir in $_UNIQUE_PREFIX_PATH; do
+  _path=$_dir/etc/ros2
   # trace output
   if [ -n "$AMENT_TRACE_SETUP_FILES" ]; then
     echo "# . \"$_path/local_setup.$AMENT_SHELL\""
@@ -127,7 +128,7 @@
   if [ -f "$_path/local_setup.$AMENT_SHELL" ]; then
     if [ "$AMENT_SHELL" = "sh" ]; then
       # provide AMENT_CURRENT_PREFIX to .sh files
-      AMENT_CURRENT_PREFIX=$_path
+      AMENT_CURRENT_PREFIX=$_dir
     fi
     # restore IFS before sourcing other files
     IFS=$_prefix_setup_IFS
@@ -136,7 +137,7 @@
     AMENT_SHELL=$_prefix_setup_AMENT_SHELL
   fi
 done
-unset _path
+unset _path _dir
 IFS=$_prefix_setup_IFS
 unset _prefix_setup_IFS
 unset _prefix_setup_AMENT_SHELL
--- ament_package/template/prefix_level/local_setup.sh.in~	2022-01-18 19:41:21.000000000 +0100
+++ ament_package/template/prefix_level/local_setup.sh.in	2022-03-30 17:48:15.129101377 +0200
@@ -103,7 +103,7 @@
 if [ "$AMENT_SHELL" != "sh" ]; then
   _ament_additional_extension="${AMENT_SHELL}"
 fi
-_ament_ordered_commands="$($_ament_python_executable "$_ament_prefix_sh_AMENT_CURRENT_PREFIX/_local_setup_util.py" sh $_ament_additional_extension)"
+_ament_ordered_commands="$($_ament_python_executable "$_ament_prefix_sh_AMENT_CURRENT_PREFIX/etc/ros2/_local_setup_util.py" sh $_ament_additional_extension)"
 unset _ament_additional_extension
 unset _ament_python_executable
 if [ -n "$AMENT_TRACE_SETUP_FILES" ]; then
--- ament_package/template/prefix_level/_local_setup_util.py~	2022-01-18 19:41:21.000000000 +0100
+++ ament_package/template/prefix_level/_local_setup_util.py	2022-04-04 22:50:55.383885104 +0200
@@ -66,7 +66,9 @@
     else:
         assert False, 'Unknown primary extension: ' + args.primary_extension
 
-    packages = get_packages(Path(__file__).parent)
+    prefix = os.path.abspath(
+        os.path.join(os.path.dirname(__file__), '..', '..'))
+    packages = get_packages(Path(prefix))
 
     ordered_packages = order_packages(packages)
     for pkg_name in ordered_packages:
@@ -74,7 +76,6 @@
             print(
                 FORMAT_STR_COMMENT_LINE.format_map(
                     {'comment': 'Package: ' + pkg_name}))
-        prefix = os.path.abspath(os.path.dirname(__file__))
         for line in get_commands(
             pkg_name, prefix, args.primary_extension,
             args.additional_extension
